source/agora/common/BlockStorage.d

```diff
@@ -382,6 +397,50 @@ public class BlockStorage
         }
     }
 
+    /***************************************************************************
+
+        Read block from the file.
+
+        Params:
+            block = `Block` to read
+            hash = `Merkle Root`
+
+        Returns:
+            Returns true if success, otherwise returns false.
+
+    ***************************************************************************/
+
+    public bool readBlockMerkleRoot (ref Block block, Hash merkle_root) @safe
+    {
+        ubyte[HASH_SIZE] hash_bytes = merkle_root[0..HASH_SIZE];
+
+        auto finds
+            = this.merkle_root_idx[].find!((a, b) => a.hash == b)(hash_bytes);
+
+        if (!finds.empty)
```

Comment:
It's better to return early. So:

```D
if (finds.empty)
    return false;
```

And then the rest of the code can be unindented by one level. It makes it easier to reason about code this way.

Reply:
Yes, I'll modify it.